File: algorithmic_toolbox/week3_greedy_algorithms/1_money_change/Change.java
import java.util.Scanner;

public class Change {
  private static int getChange(int m) {
    int n = m % 10, o = n % 5;

    return m / 10 + n / 5 + o / 1;
  }

  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    int m = scanner.nextInt();
    scanner.close();

    System.out.println(getChange(m));
  }
}

File: algorithmic_toolbox/week3_greedy_algorithms/1_money_change/change.c
#include <stdio.h>

int get_change(int m) {
  int n = m % 10, o = n % 5;

  return m / 10 + n / 5 + o / 1;
}

int main() {
  int m;
  int _ = scanf("%d", &m);
  printf("%d\n", get_change(m));
  return 0;
}
